htlcswitch/iterator.go

```diff
@@ -164,36 +166,61 @@ func (r *sphinxHopIterator) ForwardingInstructions() (ForwardingInfo, error) {
 			nextHop = lnwire.NewShortChanIDFromInt(s)
 		}
 
-		amt = lnwire.MilliSatoshi(fwdInst.ForwardAmount)
+		amt = fwdInst.ForwardAmount
 		cltv = fwdInst.OutgoingCltv
 
 	// Otherwise, if this is the TLV payload, then we'll make a new stream
 	// to decode only what we need to make routing decisions.
 	case sphinx.PayloadTLV:
-		tlvStream := tlv.NewStream(
-			tlv.MakePrimitiveRecord(tlv.AmtOnionType, &amt),
-			tlv.MakePrimitiveRecord(tlv.LockTimeOnionType, &cltv),
-			tlv.MakePrimitiveRecord(tlv.NextHopOnionType, &nextHop),
+		var cid uint64
+
+		tlvStream, err := tlv.NewStream(
+			tlv.MakeDynamicRecord(
+				tlv.AmtOnionType, &amt, nil,
+				tlv.ETUint64, tlv.DTUint64,
+			),
```

Comment:
I prefer changes to be in smaller commits if possible, as otherwise one can argue everything should just be squashed into the very same commit if it touches packages that other commits touch in the same PR.

Reply:
I am just observing that code that was introduced in this pr is rewritten here already. First you create `PrimitiveRecord` and already now they are converted to `DynamicRecord`. I think that is something to avoid if possible. Is there a reason that the changes in this function `ForwardingInstructions` couldn't have been part of the earlier commit?

Reply:
The entire PR very well could be a single commit, but IMO doing it in this way makes smaller changes in each commit. The jump here is to start to truncate the integers which is the new format in the blob.

Reply:
A single commit PR is an extreme. I am only giving the feedback that the commit structure in this particular detail confused me, that's all.